client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -343,4 +344,9 @@
     * Executes a remote script passing a set of named parameters
     */
    <T> T execute(String scriptName, Map<String, ?> params);
+
+   /**
+    * Returns segment ownership for this cache
+    */
+   Map<SocketAddress, Set<Integer>> getSegmentsByServer();
```

Comment:
Not sure I like this exposed as public API. Can't we just leave it in RemoteCacheImpl and have code that needs to use this cast to the actual implementation ?